checker/src/main/java/alluxio/checker/SparkIntegrationChecker.java

```diff
@@ -0,0 +1,243 @@
+      }
+    }
+
+    return runSparkJob(sc);
+  }
+
+  /**
+   * Spark job to check whether Spark executors can recognize Alluxio filesystem.
+   *
+   * @param sc current JavaSparkContext
+   * @return Status
+   */
+  private Status runSparkJob(JavaSparkContext sc) {
+    // Generates a list of integer for testing
+    List<Integer> nums = IntStream.rangeClosed(1, mPartitions).boxed().collect(Collectors.toList());
+
+    JavaRDD<Integer> dataSet = sc.parallelize(nums, mPartitions);
+
+    // Runs a Spark job to check whether Spark executors can recognize Alluxio
+    JavaPairRDD<Status, String> extractedStatus = dataSet
+        .mapToPair(s -> new Tuple2<>(performIntegrationChecks(), getAddress()));
+
+    // Merges the IP addresses that can/cannot recognize Alluxio
+    JavaPairRDD<Status, String> mergeStatus = extractedStatus.reduceByKey((a, b)
+        -> a.contains(b) ?  a : (b.contains(a) ? b : a + "; " + b));
+
+    mSparkJobResult = mergeStatus.collect();
+
+    // If one executor cannot recognize Alluxio, something wrong has happened
+    boolean canRecognizeClass = true;
+    boolean canRecognizeFS = true;
+    for (Tuple2<Status, String> op : mSparkJobResult) {
+      if (op._1().equals(Status.FAILCLASS)) {
+        canRecognizeClass = false;
+      }
+      if (op._1().equals(Status.FAILFS)) {
+        canRecognizeFS = false;
+      }
+    }
+    return canRecognizeClass ? (canRecognizeFS ? Status.SUCCESS : Status.FAILFS) : Status.FAILCLASS;
+  }
+
+  /**
+   * Checks if this Spark driver or executors can recognize Alluxio classes and filesystem.
+   *
+   * @return Status
+   */
+  private Status performIntegrationChecks() {
+    // Checks if Spark driver or executors can recognize Alluxio classes
+    try {
+      Class.forName("alluxio.AlluxioURI");
+      Class.forName("alluxio.client.file.BaseFileSystem");
+      Class.forName("alluxio.hadoop.AlluxioFileSystem");
+    } catch (ClassNotFoundException e) {
+      LOG.error("Failed to find Alluxio classes on classpath ", e);
+      return Status.FAILCLASS;
+    }
+
+    // Checks if Spark driver or executors can recognize Alluxio filesystem
+    try {
+      FileSystem.getFileSystemClass("alluxio", new Configuration());
+    } catch (Exception e) {
+      LOG.error("Failed to find Alluxio filesystem ", e);
+      return Status.FAILFS;
+    }
+
+    return Status.SUCCESS;
+  }
+
+  /**
+   * Gets the current Spark driver or executor IP address.
+   *
+   * @return the current Spark driver or executor IP address
+   */
+  private String getAddress() {
+    String address;
+    try {
+      address = InetAddress.getLocalHost().getHostAddress();
+    } catch (UnknownHostException e) {
+      LOG.debug("cannot get host address");
+      address = "unknown address";
+    }
+    return address;
+  }
+
+  /**
+   * Prints related Spark and Alluxio configuration information.
+   *
+   * @param conf the current SparkConf
+   */
+  private void printConfigInfo(SparkConf conf) {
```

Comment:
it would be good to propagate all information to the driver so that it can format it into a single report for presentation to the user